src/main/java/com/contentgrid/gateway/runtime/config/kubernetes/KubernetesApplicationConfigMapWatcher.java

```diff
@@ -0,0 +1,72 @@
+package com.contentgrid.gateway.runtime.config.kubernetes;
+
+import com.contentgrid.gateway.runtime.config.ComposableApplicationConfigurationRepository;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.LabelSelectorBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class KubernetesApplicationConfigMapWatcher implements AutoCloseable {
```

Comment:
I see a lot of duplication between this one and the `KubernetesApplicationSecretsWatcher`. Is it possible to make that one generic class? (I think you just need a function for `client::configMaps` or `client::secrets`